ui/src/org/ensRegistrar.ts

```diff
@@ -78,15 +78,30 @@ export function formatFee(fee: ethers.BigNumber): string {
   );
 }
 
-export async function commit(
+// Expire one hour from now.
+const deadline = ethers.BigNumber.from(Math.floor(Date.now() / 1000)).add(3600);
+
+export async function getPermitSignature(
   environment: ethereum.Environment,
-  name: string,
-  salt: Uint8Array,
   fee: ethers.BigNumber
-): Promise<{
-  receipt: ethers.providers.TransactionReceipt;
+): Promise<ethers.Signature> {
+  const wallet = svelteStore.get(Wallet.store);
+  const spender = registrarAddress(environment);
+  const token = radToken();
+  return await permitSignature(wallet.signer, token, spender, fee, deadline);
+}
+
+export interface CommitResult {
+  tx: ethers.ContractTransaction;
   minAge: number;
```

Comment:
Documentation for this would be great. I always have trouble remembering what this is for.

Reply:
69bca8ffcb5ba70807c23f35aab79f9cfb2701dd